.github/actions/slack-on-upstream-job-failure/action.yml

```diff
@@ -22,31 +22,39 @@ runs:
     - name: Get branch names
       id: branch-name
       uses: tj-actions/branch-names@v5.1
-    - name: Determine if notification is needed
-      uses: Jimdo/should-i-notify-action@d27f60fd66e60ee85b27d0a9bdf4b5fcd2abd52e
-      id: should_notify
+    - name: Get previous workflow status
+      uses: Mercymeilya/last-workflow-status@v0.3
+      id: last_status
       with:
-        branch: ${{ steps.branch-name.outputs.current_branch }}
-        needs_context: ${{ inputs.needs-context }}
         github_token: ${{ inputs.github-token }}
     - uses: actions/checkout@v2
       with:
         ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
     - name: Get short SHA
       id: vars
       shell: bash
-      run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+      run: |
+        LAST_STATUS=${{ steps.last_status.outputs.last_status }}
+        CURRENT_STATUS=${{
+          contains(fromJson(inputs.needs-context).*.result, 'failure') && 'failure'
+          || contains(fromJson(inputs.needs-context).*.result, 'cancelled') && 'cancelled'
+          || 'success'
+        }}
+        SHORT_SHA=$(git rev-parse --short HEAD)
+
+        echo "last workflow status: $LAST_STATUS"
+        echo "current workflow status: $CURRENT_STATUS"
+        echo "short SHA: $SHORT_SHA"
+
+        echo "::set-output name=last-run-status::$LAST_STATUS"
+        echo "::set-output name=this-run-status::$CURRENT_STATUS"
+        echo "::set-output name=sha_short::$SHORT_SHA"
     - name: Post to Slack App
       id: slack-app-post
-      if: ${{ steps.should_notify.outputs.should_send_message == 'yes' }}
+      if: ${{ (steps.vars.outputs.last-run-status != 'success') || (steps.vars.outputs.this-run-status != 'success') }}
```

Comment:
So now that we know that the Github API sometimes fails to deliver workflow runs, we should expect 
`steps.vars.outputs.last-run-status != 'success'` (since `'null' != 'success'`)
which in turn will trigger the slack notification. 
This is acceptable.

(The responsibility of this conditional expression is to reduce slack notifications, but it should be acceptable to temporarily have redundant notifications in slack during times when the github-api experiences issues)